src/o365/spo/commands/file/file-add.ts

```diff
@@ -554,22 +554,22 @@ class SpoFileAddCommand extends SpoCommand {
   private mapUnknownOptionsAsFieldValue(options: Options): FieldValue[] {
     const result: any = [];
     const excludeOptions: string[] = [
-      'weburl',
+      'webUrl',
       'folder',
       'path',
-      'contenttype',
-      'checkout',
-      'checkincomment',
+      'contentType',
+      'checkOut',
+      'checkInComment',
       'approve',
-      'approvecomment',
+      'approveComment',
       'publish',
-      'publishcomment',
+      'publishComment',
       'debug',
       'verbose'
     ];
 
     Object.keys(options).forEach(key => {
-      if (excludeOptions.indexOf(key.toLowerCase()) === -1) {
```

Comment:
The majority of the options are case-sensitive. We might have a few edge cases so we'll handle them on a case-by-case basis. Generally, options are case-sensitive because that's how they're passed from args to command options and that's how we handle them in validation.